File: app/src/main/java/ru/p3tr0vich/fuel/presenters/FuelingTotalPresenter.java
package ru.p3tr0vich.fuel.presenters;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.List;

import ru.p3tr0vich.fuel.models.FuelingRecord;
import ru.p3tr0vich.fuel.models.FuelingTotalModel;
import ru.p3tr0vich.fuel.views.FuelingTotalView;

public class FuelingTotalPresenter {
    private final FuelingTotalView mFuelingTotalView;
    private final FuelingTotalModel mFuelingTotalModel;

    public FuelingTotalPresenter(@NonNull FuelingTotalView fuelingTotalView) {
        mFuelingTotalView = fuelingTotalView;
        mFuelingTotalModel = new FuelingTotalModel();
        mFuelingTotalModel.setOnChangeListener(new FuelingTotalModel.OnChangeListener() {
            @Override
            public void onChange() {
                mFuelingTotalView.setAverage(mFuelingTotalModel.getAverage());
                mFuelingTotalView.setCostSum(mFuelingTotalModel.getCostSum());
            }
        });
    }

    public void onFuelingRecordsChanged(@Nullable List<FuelingRecord> fuelingRecords) {
        mFuelingTotalModel.setFuelingRecords(fuelingRecords);
    }

    public void onLastFuelingRecordsChanged(@Nullable List<FuelingRecord> fuelingRecords) {
        mFuelingTotalModel.setLastRecords(fuelingRecords);

        mFuelingTotalView.setLastConsumption(mFuelingTotalModel.getLastConsumption());
        mFuelingTotalView.setEstimatedMileage(mFuelingTotalModel.getEstimatedMileage());
        mFuelingTotalView.setEstimatedTotal(mFuelingTotalModel.getEstimatedTotal());
    }

    public void onDestroy() {
        mFuelingTotalModel.destroy();
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/presenters/FuelingTotalPresenter.kt
package ru.p3tr0vich.fuel.presenters

import ru.p3tr0vich.fuel.models.FuelingRecord
import ru.p3tr0vich.fuel.models.FuelingTotalModel
import ru.p3tr0vich.fuel.views.FuelingTotalView

class FuelingTotalPresenter(private val fuelingTotalView: FuelingTotalView) {
    private val fuelingTotalModel = FuelingTotalModel()

    init {
        fuelingTotalModel.onChangeListener = object : FuelingTotalModel.OnChangeListener {
            override fun onChange() {
                fuelingTotalView.setAverage(fuelingTotalModel.average)
                fuelingTotalView.setCostSum(fuelingTotalModel.costSum)
            }
        }
    }

    fun onFuelingRecordsChanged(fuelingRecords: List<FuelingRecord>?) {
        fuelingTotalModel.setFuelingRecords(fuelingRecords)
    }

    fun onLastFuelingRecordsChanged(fuelingRecords: List<FuelingRecord>?) {
        fuelingTotalModel.setLastRecords(fuelingRecords)

        fuelingTotalView.setLastConsumption(fuelingTotalModel.lastConsumption)
        fuelingTotalView.setEstimatedMileage(fuelingTotalModel.estimatedMileage)
        fuelingTotalView.setEstimatedTotal(fuelingTotalModel.estimatedTotal)
    }

    fun onDestroy() {
        fuelingTotalModel.destroy()
    }
}